src/main/java/seedu/toluist/model/Task.java

```diff
@@ -83,9 +89,11 @@ public boolean equals(Object other) {
                 && this.description.equals(((Task) other).description)) // state check
                 && this.priority.equals(((Task) other).priority)
                 && this.allTags.equals(((Task) other).allTags)
+                && Objects.equals(this.recurringFrequency, ((Task) other).recurringFrequency) // handles null
```

Comment:
For now lets relax this constraint bah. One possible use case is to account for like... 
CCA on every tuesday
Same CCA on every thursday
Should have equal description here but they have different dates